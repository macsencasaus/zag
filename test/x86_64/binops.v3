// RUN: %vector -stdout %s > /tmp/tmp.bin && objdump -d -M intel /tmp/tmp.bin

fn add(x: i32, y: i32): i32 {
    return x + y;
}

fn sub(x: i32, y: i32): i32 {
    return x - y;
}

fn add_then_mult(x: i64, y: i64, z: i64): i64 {
    return x + y * z;
}
