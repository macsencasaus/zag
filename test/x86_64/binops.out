
/tmp/tmp.bin:     file format elf64-x86-64


Disassembly of section .text:

0000000000000000 <add>:
   0:	55                   	push   rbp
   1:	48 89 e5             	mov    rbp,rsp
   4:	48 83 ec 10          	sub    rsp,0x10
   8:	89 7d fc             	mov    DWORD PTR [rbp-0x4],edi
   b:	89 75 f8             	mov    DWORD PTR [rbp-0x8],esi
   e:	8b 45 fc             	mov    eax,DWORD PTR [rbp-0x4]
  11:	8b 4d f8             	mov    ecx,DWORD PTR [rbp-0x8]
  14:	01 c8                	add    eax,ecx
  16:	89 45 f4             	mov    DWORD PTR [rbp-0xc],eax
  19:	8b 45 f4             	mov    eax,DWORD PTR [rbp-0xc]
  1c:	c9                   	leave
  1d:	c3                   	ret

000000000000001e <sub>:
  1e:	55                   	push   rbp
  1f:	48 89 e5             	mov    rbp,rsp
  22:	48 83 ec 10          	sub    rsp,0x10
  26:	89 7d fc             	mov    DWORD PTR [rbp-0x4],edi
  29:	89 75 f8             	mov    DWORD PTR [rbp-0x8],esi
  2c:	8b 45 fc             	mov    eax,DWORD PTR [rbp-0x4]
  2f:	8b 4d f8             	mov    ecx,DWORD PTR [rbp-0x8]
  32:	29 c8                	sub    eax,ecx
  34:	89 45 f4             	mov    DWORD PTR [rbp-0xc],eax
  37:	8b 45 f4             	mov    eax,DWORD PTR [rbp-0xc]
  3a:	c9                   	leave
  3b:	c3                   	ret

000000000000003c <add_then_mult>:
  3c:	55                   	push   rbp
  3d:	48 89 e5             	mov    rbp,rsp
  40:	48 83 ec 30          	sub    rsp,0x30
  44:	48 89 7d f8          	mov    QWORD PTR [rbp-0x8],rdi
  48:	48 89 75 f0          	mov    QWORD PTR [rbp-0x10],rsi
  4c:	48 89 55 e8          	mov    QWORD PTR [rbp-0x18],rdx
  50:	48 8b 45 f0          	mov    rax,QWORD PTR [rbp-0x10]
  54:	48 8b 4d e8          	mov    rcx,QWORD PTR [rbp-0x18]
  58:	48 f7 e9             	imul   rcx
  5b:	48 89 45 e0          	mov    QWORD PTR [rbp-0x20],rax
  5f:	48 8b 45 f8          	mov    rax,QWORD PTR [rbp-0x8]
  63:	48 8b 4d e0          	mov    rcx,QWORD PTR [rbp-0x20]
  67:	48 01 c8             	add    rax,rcx
  6a:	48 89 45 d8          	mov    QWORD PTR [rbp-0x28],rax
  6e:	48 8b 45 d8          	mov    rax,QWORD PTR [rbp-0x28]
  72:	c9                   	leave
  73:	c3                   	ret
